avro-builder/src/main/java/com/linkedin/avroutil1/builder/SchemaBuilder.java

```diff
@@ -159,6 +175,13 @@ public static void main(String[] args) throws Exception {
       handleAvro702 = Boolean.TRUE.equals(Boolean.parseBoolean(value));
     }
 
+    //allow plugins to parse and validate their own added options
+    for (BuilderPlugin plugin : plugins) {
+      plugin.parseValidateCLI(options);
```

Comment:
allowing free reign was actually my intent. there could be plugins that care about the values of some "core" CLI options perhaps?. 

i've opted to just let plugin code throw exceptions (hopefully with descriptive messages) over designing a validation API simply because its simpler.